core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1511,6 +1588,15 @@ private void handleErrorInSyncContext(Status error) {
       scheduleExponentialBackOffInSyncContext();
     }
 
+    private void drainPendingCalls() {
+      if (pendingCalls != null) {
+        for (Runnable pendingCall : pendingCalls) {
+          pendingCall.run();
+        }
+        pendingCalls = null;
```

Comment:
Is it better to offload this to the `executor`?
```java
List<Runnable> runnables = pendingCalls;
pendingCalls = null;
executor.execute(()-> {
    for (Runnable pendingCall : pendingCalls) {
      pendingCall.run();
    }});
```

Reply:
Yes, I think we need to. We should really follow DelayedClientSteram's approach: schedule each call separately in the executor and if the CallOptions had an executor, we should use it.

Reply:
Using `callExecutor` io.grpc.stub.ClientCallsTest.test blockingUnaryCall2_interruptedWaitsForOnClose is failing. Because the `callExecutor` is `ClientCalls.ThreadlessExecutor` and it throws with interrupted exception before assigning inprocess server response observer.  I think the failure is legitimate.

Reply:
> Using callExecutor io.grpc.stub.ClientCallsTest.test blockingUnaryCall2_interruptedWaitsForOnClose is failing. Because the callExecutor is ClientCalls.ThreadlessExecutor and it throws with interrupted exception before assigning inprocess server response observer. I think the failure is legitimate.

As per offline discussion, we want to keep the tests behavior because exiting users' test may also relying the currrent behavior. So we decide to make a hack/optimization at the beginning of `newCall()`.